modin/core/execution/ray/common/utils.py

```diff
@@ -186,6 +143,77 @@ def initialize_ray(
         NPartitions._put(num_cpus)
 
 
+def _object_store_memory() -> Optional[int]:
+    """
+    Get the object store memory we should start Ray with, in bytes.
+
+    If object store memory is set in the ``Memory`` Modin config variable:
+      return ``Memory``
+    else:
+      if on linux:
+        system_memory is available memory in /dev/shm
+      otherwise:
+        system_memory is psutil.virtual_memory().total
+      desired_memory = ((_OBJECT_STORE_TO_SYSTEM_MEMORY_RATIO * system_memory)
+                        rounded down to the nearest GB)
+      if desired_memory == 0:
+        return None
+      else if (on Mac and desired_memory > the Ray mac object store size
+               limit L):
+        return L
+      else:
+        return desired_memory
```

Comment:
Isn't it basically copying the function code?

I was thinking more about stuff like
```
If set by ``Memory`` config variable, return that.
On Linux, take system memory from /dev/shm, on others assume total virtual memory.
On Mac, never return more than Ray-specified upper limit.
```
Other details could be seen from the function body itself.

Reply:
This is close to the function code, but the behavior of this function for callers (not just the internals) is quite complex, so I was trying to document the behavior. I thought the description I had here was much easier to read than the code, too.

But I'm okay with what you're suggesting. Done.

Reply:
I was thinking of rules like "A docstring should give enough information to write a call to the function without reading the function’s code." from the Google Python style guide: https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

But I agree that that level of documentation does look awkward here.